test/tools/cli/IonProcess.java

```diff
@@ -387,57 +361,101 @@ private static void embeddedEventToIon(IonReader ionReader,
                             }
                         }
                         depth--;
-                        tempWriter.stepOut();
+                        embeddedContext.getIonWriter().stepOut();
                     } else if (event.getEventType() == EventType.SYMBOL_TABLE) {
-                        ionWriterForOutput.finish();
-                        ImportDescriptor[] imports = event.getImports();
-                        SymbolTable[] symbolTables = new SymbolTable[imports.length];
-                        for (int i = 0; i < imports.length; i++) {
-                            SymbolTable symbolTable = IonSystemBuilder.standard().build().newSharedSymbolTable(
-                                    imports[i].getImportName(), imports[i].getVersion(), null);
-                            symbolTables[i] = symbolTable;
-                        }
-                        ionWriterForOutput = args.getOutputFormat()
-                                .createIonWriterWithImports(outputStream, symbolTables);
+                        handleSymbolTableEvent(embeddedContext, event, outputStream, args);
                     }
                 } while (ionReader.next() != null);
 
                 if (out.length() > 0) {
-                    tempWriter.finish();
-                    ionWriterForOutput.writeString(out.toString());
+                    embeddedContext.getIonWriter().finish();
+                    processContext.getIonWriter().writeString(out.toString());
+                }
+            } finally {
+                IonWriter ionWriter = embeddedContext.getIonWriter();
+                if (ionWriter != null) {
+                    try {
+                        ionWriter.close();
+                    } catch (IOException e) {
+                        System.err.println(e.getMessage());
```

Comment:
You can't keep going after this, right? I think this exception should either be allowed to bubble up, or you should exit from here.